### PR TITLE
Always sync and log push tokens. Apply OWSAlerts in more places.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/AdvancedSettingsTableViewController.m [hunk 1]
```diff
@@ -130,21 +129,9 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
-                                               accountManager:[Environment getCurrent].accountManager
-                                                  preferences:[Environment preferences]];
-        syncJob.uploadOnlyIfStale = NO;
-        [syncJob run]
-            .then(^{
-                DDLogDebug(@"%@ Successfully ran syncPushTokensJob.", self.tag);
-                SignalAlertView(NSLocalizedString(@"PUSH_REGISTER_SUCCESS", @"Alert title"), nil);
-            })
-            .catch(^(NSError *error) {
-                DDLogError(@"%@ Failed to run syncPushTokensJob with error: %@", self.tag, error);
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/AdvancedSettingsTableViewController.m]: What you have now is consolidating code nicely, but I liked giving the user visual feedback here, while just logging it elsewhere. Reason being that this is in response to an *explicit* action the user takes.

e.g. How does the user know "Did it work?"

File: Signal/Signal-Prefix.pch [hunk 2]
```diff
@@ -20,6 +20,4 @@
     #import <SignalServiceKit/OWSAnalytics.h>
     #import <SignalServiceKit/OWSDispatch.h>
     #import <SignalServiceKit/iOSVersions.h>
-
-    #define SignalAlertView(title,msg) [[[UIAlertView alloc] initWithTitle:title message:msg delegate:nil cancelButtonTitle:NSLocalizedString(@"OK", @"") otherButtonTitles:nil, nil] show]
```
Comment (michaelkirk) [inline Signal/Signal-Prefix.pch]: so long deprecated code 👋

File: Signal/src/views/OWSAlerts.swift [hunk 3]
```diff
@@ -23,15 +23,18 @@ import Foundation
         UIApplication.shared.frontmostViewController?.present(alertController, animated: true, completion: nil)
     }
 
+    public class func showAlert(withTitle title: String) {
+        self.showAlert(withTitle: title, message: nil, buttonTitle: NSLocalizedString("OK", comment: ""))
+    }
+
     public class func showAlert(withTitle title: String, message: String) {
```
Comment (michaelkirk) [inline Signal/src/views/OWSAlerts.swift]: nit: default params might dedupe this helpfully

https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Functions.html#//apple_ref/doc/uid/TP40014097-CH10-ID169

File: Signal/src/ViewControllers/AdvancedSettingsTableViewController.m (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/AdvancedSettingsTableViewController.m]: Restored!

File: Signal/src/views/OWSAlerts.swift (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/views/OWSAlerts.swift]: Well, I've done this but we're almost entirely using this from Objective-C and you can't use default params from Objective-C - at least not to my knowledge.

**Pull request merged**